core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -72,21 +96,55 @@ object ClientConfigOverride {
- * The callback provides the clientId and the full properties set read from ZK.
- * This implementation reports the overrides to the respective ClientQuotaManager objects
- */
-class ClientIdConfigHandler(private val quotaManagers: Map[Short, ClientQuotaManager]) extends ConfigHandler {
+  * The ClientIdConfigHandler will process clientId config changes in ZK.
+  * The callback provides the clientId and the full properties set read from ZK.
+  * This implementation reports the overrides to the respective ClientQuotaManager objects
+  */
+class ClientIdConfigHandler(private val quotaManagers: QuotaManagers) extends ConfigHandler {
 
-  def processConfigChanges(clientId: String, clientConfig: Properties) = {
+  def processConfigChanges(clientId: String, clientConfig: Properties) {
     if (clientConfig.containsKey(ClientConfigOverride.ProducerOverride)) {
-      quotaManagers(ApiKeys.PRODUCE.id).updateQuota(clientId,
+      quotaManagers.client(QuotaType.Produce).updateQuota(clientId,
         new Quota(clientConfig.getProperty(ClientConfigOverride.ProducerOverride).toLong, true))
     }
 
     if (clientConfig.containsKey(ClientConfigOverride.ConsumerOverride)) {
-      quotaManagers(ApiKeys.FETCH.id).updateQuota(clientId,
+      quotaManagers.client(QuotaType.Fetch).updateQuota(clientId,
         new Quota(clientConfig.getProperty(ClientConfigOverride.ConsumerOverride).toLong, true))
     }
   }
 }
+
+/**
+  * The BrokerConfigHandler will process individual broker config changes in ZK.
+  * The callback provides the brokerId and the full properties set read from ZK.
+  * This implementation reports the overrides to the respective ReplicationQuotaManager objects
+  */
+class BrokerConfigHandler(private val config: KafkaConfig, private val quotaManagers: QuotaManagers) extends ConfigHandler with Logging{
+
+  def processConfigChanges(proposedConfigChangeBrokerId: String, proposedProperties: Properties) {
+    if(config.brokerId == proposedConfigChangeBrokerId.trim.toInt) {
+      val limit = proposedProperties.getProperty(KafkaConfig.ThrottledReplicationRateLimitProp).toLong
+      config.mutateConfig(KafkaConfig.ThrottledReplicationRateLimitProp, limit)
+      if (proposedProperties.containsKey(KafkaConfig.ThrottledReplicationRateLimitProp)) {
+        quotaManagers.leaderReplication.updateQuota(new Quota(limit, true))
+        quotaManagers.followerReplication.updateQuota(new Quota(limit, true))
+      }
+    }
+  }
+}
+
+object ThrottledReplicaValidator extends Validator {
+  override def ensureValid(name: String, value: scala.Any): Unit = {
+    if (!value.isInstanceOf[String])
+      throw new ConfigException(name, value, s"$name  must be a string")
+    if (!isValid(value.toString)) {
+      throw new ConfigException(name, value, s"$name  must match for format [number]-[number]:[number]-[number]:[number]-[number] etc")
+    }
+  }
+
+  def isValid(value: String): Boolean = {
```

Comment:
Yeah - it was just used by a test. Have changed to test the ensureValid method instead.